sdk/resourcemanager/azure-resourcemanager-samples/src/main/java/com/azure/resourcemanager/compute/samples/CreateVirtualMachineEncryptedUsingCustomerManagedKey.java

```diff
@@ -0,0 +1,207 @@
+        final String password = Utils.password();
+        final String sshPublicKey = Utils.sshPublicKey();
+        try {
+
+            //=============================================================
+            // Create key vault and key
+
+            Vault vault = azureResourceManager.vaults()
+                .define(vaultName)
+                .withRegion(region)
+                .withNewResourceGroup(rgName)
+                .defineAccessPolicy()
+                    .forServicePrincipal(clientId)
+                    .allowKeyPermissions(KeyPermissions.CREATE)
+                    .attach()
+                .withPurgeProtectionEnabled()
+                .create();
+
+            Key vaultKey = vault.keys()
+                .define(keyName)
+                .withKeyTypeToCreate(KeyType.RSA)
+                .withKeySize(4096)
+                .create();
+
+            //=============================================================
+            // Create disk encryption set
+
+            DiskEncryptionSet des = azureResourceManager.diskEncryptionSets()
+                .define(desName)
+                .withRegion(region)
+                .withExistingResourceGroup(rgName)
+                .withEncryptionType(DiskEncryptionSetType.ENCRYPTION_AT_REST_WITH_CUSTOMER_KEY)
+                .withExistingKeyVault(vault.id())
+                .withExistingKey(vaultKey.id())
+                .withSystemAssignedManagedServiceIdentity()
+                .withAutomaticKeyRotation()
+                .create();
+
+            //=============================================================
+            // Grant disk encryption set access to the key vault by defining key vault access policy
+
+            vault.update()
+                .defineAccessPolicy()
+                    .forObjectId(des.systemAssignedManagedServiceIdentityPrincipalId())
+                    .allowKeyPermissions(KeyPermissions.GET, KeyPermissions.WRAP_KEY, KeyPermissions.UNWRAP_KEY)
+                    .attach()
+                .apply();
+
+            //=============================================================
+            // Create virtual machine with OS disk encrypted using customer managed keys and one data disk lun1 encrypted using
+            // platform-managed key
+
+            VirtualMachine linuxVM = azureResourceManager.virtualMachines()
+                .define(vmName)
+                .withRegion(region)
+                .withExistingResourceGroup(rgName)
+                .withNewPrimaryNetwork("10.0.0.0/28")
+                .withPrimaryPrivateIPAddressDynamic()
+                .withoutPrimaryPublicIPAddress()
+                .withPopularLinuxImage(KnownLinuxVirtualMachineImage.UBUNTU_SERVER_16_04_LTS)
+                .withRootUsername("jvuser")
+                .withRootPassword(password)
+                .withSsh(sshPublicKey)
+                .withNewDataDisk(10, 1, new VirtualMachineDiskOptions().withDiskEncryptionSet(null))
```

Comment:
I assume user do not need `new VirtualMachineDiskOptions().withDiskEncryptionSet(null)`, if they do not provide default encrytion setting to data disks?

Reply:
Right, I'll change it to other overload method.